qa/L0_trace/opentelemetry_unittest.py

```diff
@@ -153,121 +191,560 @@ def _check_events(self, span_name, events):
-        # Order of spans and their relationship described earlier
-        for child, parent in zip(self.spans[3:6], self.spans[4:6]):
-            self._check_parent(child, parent)
-
-        # Next 4 spans in `self.spans` belong to ensemble request
-        # Order of spans: compute span - request span - request span - root span
-        for child, parent in zip(self.spans[6:10], self.spans[7:10]):
-            self._check_parent(child, parent)
-
-        # Final 6 spans in `self.spans` belong to bls with ensemble request
-        # Order of spans:
-        # compute span - request span (simple) - request span (ensemble)-
-        # - compute (for bls) - request (bls) - root span
-        # request span (ensemble) and compute (for bls) are children of
-        # request (bls)
-        children = self.spans[10:]
-        parents = (self.spans[11:13], self.spans[14], self.spans[14:])
-        for child, parent in zip(children, parents[0]):
-            self._check_parent(child, parent)
-
-    def test_resource_attributes(self):
-        for span in self.spans:
-            self.assertIn("test.key", span["resources"])
-            self.assertEqual("test.value", span["resources"]["test.key"])
-            self.assertIn("service.name", span["resources"])
-            self.assertEqual("test_triton", span["resources"]["service.name"])
-
-
-def prepare_data(client):
-    inputs = []
-    input0_data = np.full(shape=(1, 16), fill_value=-1, dtype=np.int32)
-    input1_data = np.full(shape=(1, 16), fill_value=-1, dtype=np.int32)
+            len(span_names),
+            sum(expected_counts.values()),
+            "Unexpeced number of span names collected",
+        )
+        for name, count in expected_counts.items():
+            self.assertEqual(
+                span_names.count(name),
+                count,
+                "Unexpeced number of " + name + " spans collected",
+            )
 
-    inputs.append(client.InferInput("INPUT0", [1, 16], "INT32"))
-    inputs.append(client.InferInput("INPUT1", [1, 16], "INT32"))
+    def _verify_nesting(self, spans, expected_parent_span_dict):
+        """
+        Helper function that checks parent-child relationships between
+        collected spans are the same as in `expected_parent_span_dict`.
+
+        Args:
+            spans (List[dict]): list of json objects, extracted from the trace and
+                   containing span info. For this test `name`
+                   and `events` are required.
+            expected_parent_span_dict (dict): dictionary, containing expected
+                   parents and children in the dictionary form:
+                        <parent_span_name> (str) : <children_names> (List[str])
+        """
+        seen_spans = dict({})
+        for span in spans:
+            cur_span = span[0]["spanId"]
+            seen_spans[cur_span] = span[0]["name"]
+
+        parent_child_dict = dict({})
+        for span in spans:
```

Comment:
Any reason for this `dict({})` syntax over `{}` or `dict()`?
```suggestion
        seen_spans = {}
        for span in spans:
            cur_span = span[0]["spanId"]
            seen_spans[cur_span] = span[0]["name"]

        parent_child_dict = {}
        for span in spans:
```

Reply:
Nicely spotted. I can incorporate this
No particular reason in this case